pal/src/httpapi_sl.c

```diff
@@ -0,0 +1,492 @@
+    }
+}
+
+static int splitHeader(char *headerName, char **headerValue)
+{
+    *headerValue = strchr(headerName, ':');
+    if (*headerValue == NULL) {
+         return (-1);
+    }
+
+    **headerValue = '\0';
+    (*headerValue)++;
+    while (**headerValue == ' ') {
+        (*headerValue)++;
+    }
+
+    return (0);
+}
+
+HTTPAPI_RESULT HTTPAPI_Init(void)
+{
+    return (HTTPAPI_OK);
+}
+
+void HTTPAPI_Deinit(void)
+{
+}
+
+HTTP_HANDLE HTTPAPI_CreateConnection(const char* hostName)
+{
+    bool error = false;
+    HTTPAPI_Object * apiH;
+    int16_t statusCode;
+
+    apiH = calloc(sizeof(HTTPAPI_Object), 1);
+    if (!apiH) {
+        LogError("Error creating HTTPAPI object\n");
+        error = true;
+        goto error;
+    }
+
+    apiH->cli = HTTPClient_create(&statusCode, 0);
+    if (statusCode < 0) {
+        LogError("Error creating http client\n");
+        error = true;
+        goto error;
+    }
+
+    apiH->prefixedHostName = malloc(strlen(hostName) + 9);  /* https:// */
+    if (!apiH->prefixedHostName) {
+        LogError("Error creating hostname buffer\n");
+        error = true;
+        goto error;
+    }
+    /*
+     * Microsoft is assuming we would connect through the secured port.
+     * We need to drop a hint for HTTPClient_connect()
+     */
+    strcpy(apiH->prefixedHostName, "https://");
+    strcat(apiH->prefixedHostName, hostName);
+
+error:
+    if (error) {
+        if (apiH->cli != NULL) {
+            HTTPClient_destroy(apiH->cli);
+        }
+        if (apiH->prefixedHostName != NULL) {
+            free(apiH->prefixedHostName);
+        }
+        if (apiH != NULL) {
+            free(apiH);
+            apiH = NULL;
+        }
+    }
+
+    return ((HTTP_HANDLE)apiH);
+}
+
+void HTTPAPI_CloseConnection(HTTP_HANDLE handle)
+{
+    HTTPAPI_Object * apiH = (HTTPAPI_Object *)handle;
+
+    if (apiH->cli != NULL) {
+        if (apiH->isConnected) {
+            HTTPClient_disconnect(apiH->cli);
+        }
+        if (apiH->prefixedHostName != NULL) {
+            free(apiH->prefixedHostName);
+        }
+        if (apiH->x509Certificate != NULL) {
+            free(apiH->x509Certificate);
+        }
+        if (apiH->x509PrivateKey != NULL) {
+            free(apiH->x509PrivateKey);
+        }
+        HTTPClient_destroy(apiH->cli);
+    }
+
+    if (apiH != NULL) {
+        free(apiH);
+    }
+}
+
+HTTPAPI_RESULT HTTPAPI_ExecuteRequest(HTTP_HANDLE handle,
+        HTTPAPI_REQUEST_TYPE requestType, const char* relativePath,
+        HTTP_HEADERS_HANDLE httpHeadersHandle, const unsigned char* content,
+        size_t contentLength, unsigned int* statusCode,
+        HTTP_HEADERS_HANDLE responseHeadersHandle,
+        BUFFER_HANDLE responseContent)
+{
+    HTTPAPI_Object * apiH = (HTTPAPI_Object *)handle;
+    HTTPClient_Handle cli = apiH->cli;
+    int i;
+    int ret;
+    int offset;
+    size_t cnt;
+    char contentBuf[CONTENT_BUF_LEN] = {0};
+    uint32_t contentBufLen = CONTENT_BUF_LEN;
```

Comment:
Is there any danger of this being too much data to allocate on the stack? It's 1280 bytes with the current definition.